BeagleBone_black/src/communications/communications.cpp

```diff
@@ -56,13 +56,10 @@ Communications::Communications(Logger& log, const char* ip, int portNo)
 
   if (connect(sockfd_, (struct sockaddr *) &serv_addr, sizeof(serv_addr)) < 0) {
     log_.ERR("COMN", "CANNOT ESTABLISH CONNECTION TO BASE-STATION.");
+  } else {
+    log_.INFO("COMN", "TCP/IP connection established.");
+    connected_ = true;
```

Comment:
Are you sure this is the right way around? Does `connect` return true on failure and false on success? Maybe add a comment explaining the oddity.

Reply:
added an explaining comment